contracts/test/CompoundGovernorSetWhitelistAccountExpiration.t.sol

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity 0.8.26;
+
+import {ProposalTest} from "contracts/test/helpers/ProposalTest.sol";
+import {IGovernor} from "@openzeppelin/contracts/governance/IGovernor.sol";
+import {CompoundGovernor} from "contracts/CompoundGovernor.sol";
+
+contract CompoundGovernorSetWhitelistAccountExpirationTest is ProposalTest {
+    function testFuzz_WhitelistAnAccountAsTimelock(address _account, uint256 _expiration) public {
+        vm.prank(TIMELOCK_ADDRESS);
+        governor.setWhitelistAccountExpiration(_account, _expiration);
```

Comment:
just making sure i understand, we're not passing a proposal here because we don't use `_checkGovernance` here, correct?

Reply:
you raise a good point actually. The current check we have is 
```ts
       if (msg.sender != _executor() && msg.sender != whitelistGuardian) {
            revert Unauthorized("Not timelock or guardian", msg.sender);
        }
```

I can it to be like

```
    if(msg.sender != whitelistGuardian) {
    _checkGovernance();
    }
```

But then we will only get the error `GovernorOnlyExecutor`, instead one that indicates that the msg.sender can also be a whitelistGuardian. What do you think?

Reply:
i'm fine leaving this as is